stackslib/src/chainstate/stacks/boot/signers_voting_tests.rs

```diff
@@ -0,0 +1,433 @@
+/// Alice can vote successfully.
+/// Bob is out of the voting window.
+#[test]
+fn vote_for_aggregate_public_key_in_last_block() {
+    let stacker_1 = TestStacker::from_seed(&[3, 4]);
+    let stacker_2 = TestStacker::from_seed(&[5, 6]);
+    let observer = TestEventObserver::new();
+
+    let signer_1 = key_to_stacks_addr(&stacker_1.signer_private_key).to_account_principal();
+    let signer_2 = key_to_stacks_addr(&stacker_2.signer_private_key).to_account_principal();
+
+    let (mut peer, mut test_signers, latest_block_id, current_reward_cycle) = prepare_signers_test(
+        function_name!(),
+        vec![(signer_1, 1000), (signer_2, 1000)],
+        Some(vec![&stacker_1, &stacker_2]),
+        Some(&observer),
+    );
+
+    let mut stacker_1_nonce: u64 = 1;
+    let dummy_tx_1 = make_dummy_tx(
+        &mut peer,
+        &stacker_1.stacker_private_key,
+        &mut stacker_1_nonce,
+    );
+    let dummy_tx_2 = make_dummy_tx(
+        &mut peer,
+        &stacker_1.stacker_private_key,
+        &mut stacker_1_nonce,
+    );
+    let dummy_tx_3 = make_dummy_tx(
+        &mut peer,
+        &stacker_1.stacker_private_key,
+        &mut stacker_1_nonce,
+    );
+
+    let cycle_id: u128 = current_reward_cycle;
+    let aggregated_public_key: Point = Point::new();
+
+    // create vote txs for alice
+    let signer_1_nonce = 0;
+    let signer_1_key = &stacker_1.signer_private_key;
+    let signer_1_address = key_to_stacks_addr(signer_1_key);
+    let signer_1_principal = PrincipalData::from(signer_1_address);
+    let signer_1_index = get_signer_index(&mut peer, latest_block_id, signer_1_address);
+
+    let txs_1 = vec![
+        // cast a vote for the aggregate public key
+        make_signers_vote_for_aggregate_public_key(
+            signer_1_key,
+            signer_1_nonce,
+            signer_1_index,
+            &aggregated_public_key,
+            0,
+        ),
+        // cast the vote twice
+        make_signers_vote_for_aggregate_public_key(
+            signer_1_key,
+            signer_1_nonce + 1,
+            signer_1_index,
+            &aggregated_public_key,
+            0,
+        ),
+    ];
+
+    // create vote txs for bob
+    let signer_2_nonce = 0;
+    let signer_2_key = &stacker_2.signer_private_key;
+    let signer_2_address = key_to_stacks_addr(signer_2_key);
+    let signer_2_principal = PrincipalData::from(signer_2_address);
+    let signer_2_index = get_signer_index(&mut peer, latest_block_id, signer_2_address);
+
+    let txs_2 = vec![
+        // cast a vote for the aggregate public key
+        make_signers_vote_for_aggregate_public_key(
+            signer_2_key,
+            signer_2_nonce,
+            signer_2_index,
+            &aggregated_public_key,
+            0,
+        ),
+    ];
+
+    //
+    // vote in the last burn block of prepare phase
+    //
+
+    nakamoto_tenure(
+        &mut peer,
+        &mut test_signers,
+        vec![vec![dummy_tx_1]],
+        signer_1_key,
+    );
```

Comment:
Can be worth it adding a helper to advance _N_ burn blocks. (And construct the dummy tx inside.)

Reply:
:100: 

However, tooling for rust testing pox is out of scope of this PR.

Currently, I like the explicit advancement.

Reply:
Fair enough!